pkg/auth/user/UserCommonService.go

```diff
@@ -329,24 +330,60 @@ func (impl UserCommonServiceImpl) RemoveRolesAndReturnEliminatedPolicies(userInf
 	// delete remaining Ids from casbin role mapping table in orchestrator and casbin policy db
 	// which are existing but not provided in this request
 	eliminatedRoles := make([]*repository.RoleModel, 0, len(eliminatedRoleIds))
+	roleIdVsRoleMap, err := impl.getMapOfRoleIdVsRoleForRoleIds(maps.Keys(eliminatedRoleIds))
+	if err != nil {
+		impl.logger.Errorw("error encountered in RemoveRolesAndReturnEliminatedPolicies", "eliminatedRoleIds", eliminatedRoleIds, "err", err)
+		return nil, nil, err
+	}
+	toBeDeletedUserRolesIds := make([]int, 0, len(eliminatedRoleIds))
 	for _, userRoleModel := range eliminatedRoleIds {
-		role, err := impl.userAuthRepository.GetRoleById(userRoleModel.RoleId)
-		if err != nil {
-			return nil, nil, err
+		if role, ok := roleIdVsRoleMap[userRoleModel.RoleId]; ok {
+			if len(role.Team) > 0 {
+				rbacObject := fmt.Sprintf("%s", role.Team)
+				isValidAuth := managerAuth(casbin.ResourceUser, token, rbacObject)
+				if !isValidAuth {
+					continue
+				}
```

Comment:
will this flow be usde for both helm and devtron apps? if yes, helm apps rbac should be derived through superadmin permission

Reply:
Have Extracted out wholelogic